### PR TITLE
Update user guide.

### PR DESCRIPTION
**Pull request by pelesh**
Reorganizing ReSolve's documentation. This is work in progress; the PR has been created to provide preview to other developers and discuss how we want to organize docs.

- Doxygen generated docs with modifications from this PR can be previewed [here](https://resolve.readthedocs.io/en/user-doc-dev/doxygen/html/index.html).
- Sphyinx generated docs with modifications from this PR are available [here](https://resolve.readthedocs.io/en/user-doc-dev/).

### CODE REVIEW

File: docs/sphinx/user_guide/index.rst [hunk 1]
```diff
@@ -3,12 +3,17 @@ User Guide
...
-Dependencies: - KLU, AMD and COLAMD libraries from SuiteSparse - CUDA >= 11.4 - CMake >= 3.22
+* C++11 compliant compiler
+* CMake >= 3.22
+* KLU, AMD and COLAMD libraries from SuiteSparse >= 5.0 (optional)
+* CUDA >= 11.4 (optional)
+* HIP/ROCm >= 5.6 (optional)
 
-To build it:
+
+To acquire and build Re::Soleve it is as easy as:
```
Comment (ryandanehy) [inline docs/sphinx/user_guide/index.rst:16]: Typo

File: docs/index.rst [hunk 2]
```diff
@@ -2,41 +2,53 @@
...
 
 =============
 Documentation
 =============
 
-Source code documentation are also linked on this site.
-`ReSolve Github Project <https://github.com/ORNL/ReSolve>`_.
-`Source documentation <doxygen/html/index.html>`_
+To get started, please check our `User Guide <sphinx/user_guide/index.html>`_
+Source code documentation generated in Doxygen is also
```
Comment (ryandanehy) [inline docs/index.rst:17]: weird formatting also can be seen here https://resolve.readthedocs.io/en/user-doc-dev/

File: docs/index.rst [hunk 3]
```diff
@@ -2,41 +2,53 @@
...
+`linked <doxygen/html/index.html>`_ to this site.
+
 
 Contributing
 ------------
 
 For all contributions to ReSolve please follow the `developer
-guidelines <sphinx/coding_guide/index.html>`__
+guidelines <sphinx/coding_guidelines/index.html>`__
```
Comment (ryandanehy) [inline docs/index.rst:25]: This takes us to coding style page. Maybe it should take us to the developer guide instead?

**Review by ryandanehy — APPROVED** [review]
I have some small suggestions that I left as comments but looks good.

File: docs/index.rst (hunk 2 above)
Comment (pelesh) [reply docs/index.rst:17]: I am not sure I know what you are referring to here, but I did make some minor formatting changes. Please check.

File: docs/index.rst (hunk 3 above)
Comment (pelesh) [reply docs/index.rst:25]: This is the same behavior as before, I just renamed coding guide to coding guidelines.

Comment (pelesh) [reply docs/index.rst]: Made some changes per @ryandanehy suggestions.

File: docs/sphinx/user_guide/index.rst (hunk 1 above)
Comment (pelesh) [reply docs/sphinx/user_guide/index.rst]: Fixed.

**Pull request merged**